GoogleUtilities/Logger/GULLogger.m

```diff
@@ -172,12 +162,32 @@ void GULLogBasic(GULLoggerLevel level,
   } else {
     logMsg = [[NSString alloc] initWithFormat:message arguments:args_ptr];
   }
-  logMsg = [NSString stringWithFormat:@"%@ - %@[%@] %@", sVersion, service, messageCode, logMsg];
+  logMsg = [NSString stringWithFormat:@"%@ - %@[%@] %@", sVersion, category, messageCode, logMsg];
   dispatch_async(sGULClientQueue, ^{
-    asl_log(sGULLoggerClient, NULL, (int)level, "%s", logMsg.UTF8String);
+    NSMutableDictionary<GULLoggerService, os_log_t> *subsystemLogs = sGULServiceLogs[subsystem];
+    if (!subsystemLogs) {
+      subsystemLogs = [NSMutableDictionary dictionary];
+      sGULServiceLogs[subsystem] = subsystemLogs;
+    }
+
+    os_log_t serviceLog = [subsystemLogs objectForKey:subsystem];
+    if (!serviceLog) {
+      serviceLog = os_log_create(subsystem.UTF8String, category.UTF8String);
+      subsystemLogs[category] = serviceLog;
+    }
+
+    os_log_with_type(serviceLog, GULLoggerLevelToOSLogType(level), "%{public}@", logMsg);
```

Comment:
what is `"%{public}@"`?

Reply:
https://developer.apple.com/documentation/os/logging/generating_log_messages_from_your_code?language=objc#3665948